### ISSUE TITLE
Problems with .sqlite files

### ISSUE THREAD

**Issue by kokorap27** [r-spatial/sf#778]
Hola!

I have been struggling with reading a .sqlite file with **sf** -CORINE Land Cover 2012, available at https://land.copernicus.eu/pan-european/corine-land-cover/clc-2012 - which I did quite straightforward with R 3.4.4 on Ubuntu 16.04. 

Before install gdal 2.3.0 from source, which I thought it was the problem because st_layers was working, my R session just aborted and now I got the following error messages:

st_layers("./CLC2012/clc12_Version_18_5.sqlite")
Cannot open data source /home/koko/Dropbox/UofExeter/Landfills/Data/CLC201/clc12_Version_18_5.sqlite
Error in CPL_get_layers(dsn, options, do_count) : Open failed. 

CLC<-st_read("./CLC2012/clc12_Version_18_5.sqlite","clc12_Version_18_5")
Cannot open data source /home/koko/Dropbox/UofExeter/Landfills/Data/CLC201/clc12_Version_18_5.sqlite
Error in CPL_read_ogr(dsn, layer, as.character(options), quiet, type,  : Open failed.

Do am I missing something related with package installation?

Here is some session info:

> library(sf)
Linking to GEOS 3.6.2, GDAL 2.3.0, proj.4 4.9.3

R version 3.5.0 (2018-04-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04 LTS

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

Best,
Roman.

**Comment by edzer** [comment]
I see
```r
> library(sf)
Linking to GEOS 3.6.2, GDAL 2.2.3, proj.4 4.9.3
> clc = st_read("clc12_Version_18_5.sqlite")
Reading layer `clc12_version_18_5' from data source `/tmp/clc12_Version_18_5.sqlite' using driver `SQLite'
Simple feature collection with 2370829 features and 4 fields
geometry type:  MULTIPOLYGON
dimension:      XYZM
bbox:           xmin: -2693292 ymin: -3086662 xmax: 10037210 ymax: 5440568
epsg (SRID):    3035
proj4string:    +proj=laea +lat_0=52 +lon_0=10 +x_0=4321000 +y_0=3210000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs
> head(clc)
Simple feature collection with 6 features and 4 fields
geometry type:  MULTIPOLYGON
dimension:      XYZM
bbox:           xmin: -2620304 ymin: 652677.7 xmax: -2474183 ymax: 1028210
epsg (SRID):    3035
proj4string:    +proj=laea +lat_0=52 +lon_0=10 +x_0=4321000 +y_0=3210000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs
  code_12   ID Remark      Area_Ha                          Shape
1     523 EU-1   <NA>     32.51698 MULTIPOLYGON ZM (((-2577067...
2     523 EU-2   <NA> 404899.16478 MULTIPOLYGON ZM (((-2605899...
3     111 EU-3   <NA>     78.16364 MULTIPOLYGON ZM (((-2516165...
4     112 EU-4   <NA>     75.16574 MULTIPOLYGON ZM (((-2514434...
5     112 EU-5   <NA>     46.83988 MULTIPOLYGON ZM (((-2515373...
6     112 EU-6   <NA>     54.54922 MULTIPOLYGON ZM (((-2525634...
```
but note that the process required 44 Gb of memory to work: does your machine have that? It also seems like a waste of space to me that this sqlite comes as XYZM geometry, as Z and M seem to be unused:
```r
> clc$Shape[[1]]
MULTIPOLYGON ZM (((-2577067 949516.6 0 NaN, -2577070 949499.7 0 NaN, -2577155 949592.5 0 NaN, -2577255 949763.2 0 NaN, -2577358 949915.6 0 NaN, -2577467 950067.4 0 NaN, -2577552 950167.4 0 NaN, -2577633 950222.6 0 NaN, -2577693 950261.9 0 NaN, -2577702 950333.2 0 NaN, -2577738 950397.4 0 NaN, -2577786 950462.3 0 NaN, -2577784 950465 0 NaN, -2577955 950752.2 0 NaN, -2577963 950753 0 NaN, -2577980 950757.3 0 NaN, -2577996 950771.1 0 NaN, -2578013 950815.5 0 NaN, -2578013 950837.7 0 NaN, -2578011 950846.5 0 NaN, -2578033 950882.9 0 NaN, -2578181 951123.9 0 NaN, -2578323 951318.9 0 NaN, -2578392 951411.9 0 NaN, -2578417 951486.1 0 NaN, -2578431 951547.9 0 NaN, -2578546 951699.1 0 NaN, -2578626 951835.7 0 NaN, -2578723 951988.7 0 NaN, -2578921 952321.3 0 NaN, -2579072 952550.2 0 NaN, -2579213 952802.7 0 NaN, -2579309 952928.7 0 NaN, -2579366 952986.3 0 NaN, -2579422 953111.7 0 NaN, -2579436 953250.3 0 NaN, -2579392 953216.7 0 NaN, -2579333 953157 0 NaN, -2579262 953065.4 0 NaN, -2578893 952505.7 0 NaN, -2578689 952105.9 0 NaN, -2578521 951797.5 0 NaN, -2578373 951536.8 0 NaN, -2578318 951458.9 0 NaN, -2578270 951397.9 0 NaN, -2578129 951172.5 0 NaN, -2577987 950911.2 0 NaN, -2577812 950630.5 0 NaN, -2577704 950438 0 NaN, -2577644 950322.1 0 NaN, -2577595 950272.7 0 NaN, -2577530 950193.2 0 NaN, -2577332 949950.9 0 NaN, -2577211 949773.2 0 NaN, -2577067 949516.6 0 NaN)))
```
Isn't it easier to process these data as raster, using package `raster` on the GeoTIFF file?

**Comment by kokorap27** [comment]
Thanks a lot for replying me quickly.

So should I remove gdal and re-install it using the ppa ubuntugis-unstable as you suggests? Actually, I reinstalling Ubuntu 18.04 in order to make a clear and minimal installation of R so any suggestion is more than welcome and appreciate in advance. 

Gosh! Last time with R 3.4.4 on Ubuntu 16.04 it took me for reading it around 4-5 minutes and 15 Gb of the R memory (by the way, I have 32 Gb of memory). And yes, I'm using  st_zm(clc) %>% select(ID,code_12) to save space - ending up with a 8.6 Gb file.

Regarding the GeoTIFF file, I understand that it's cheaper and easier for reading these data but I don't have idea to what extend this option will work better - I'm disaggregating population so I think that the resolution is really important. Again, any guidance or resource for learning more about it is more than welcome.

**Comment by edzer** [comment]
As of gdal versions: yes, I use 18.04, but with native gdal, I believe this is without ubuntugis-untsable (although that shouldn't matter?).

**Comment by kokorap27** [comment]
Right! I already reinstalled Ubuntu 18.04, R 3.5.0 and native gdal and now **st_read** it's working for .sqlite files- I used a smaller size file (I should try this before ask). Thanks a lot for supporting me.

Nevertheless, I think there's something important on why I could read the CLC file when using Ubuntu 16.04 with R. 3.4.4 without memory problems on the same computer. Any clue would be really useful in order to boost the capacity of R.
